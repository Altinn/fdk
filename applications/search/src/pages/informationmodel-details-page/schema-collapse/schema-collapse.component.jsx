import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import _ from 'lodash';
import { Collapse } from 'reactstrap';
import { withStateHandlers } from 'recompose';

import './schema-collapse.scss';

export const SchemaCollapsePure = props => {
  const { title, type, marginLeft, nestLevel, onToggle, collapse } = props;

  if (!(title || type)) {
    return null;
  }

  const collapseIconClass = cx(
    'fa',
    'font-weight-bold',
    'mr-2',
    'fdk-color-blue-dark',
    {
      'fa-angle-down': !collapse,
      'fa-angle-up': collapse
    }
  );

  return (
    <React.Fragment>
      {nestLevel === 1 && (
        <React.Fragment>
          <button
            style={{ marginLeft }}
            className="bg-transparent border-0 p-0 w-100"
            onClick={onToggle}
            name={title}
          >
            <div className="d-flex align-items-center">
              <div className="d-flex justify-content-start w-50">
                <i className={collapseIconClass} />
                <span className="fdk-text-extra-strong">{title}</span>
              </div>
              <span className="d-flex w-50">{type}</span>
            </div>
          </button>
          <Collapse className="mt-2" isOpen={collapse}>
            {props.children}
          </Collapse>
        </React.Fragment>
      )}
      {nestLevel > 1 && (
        <React.Fragment>
          <div style={{ marginLeft }} className="d-flex schema-collapse--item">
            <span className="w-50">{title}</span>
            <span className="w-50">{type}</span>
          </div>
          {props.children}
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

SchemaCollapsePure.defaultProps = {
  title: null,
  type: null,
  marginLeft: null,
  nestLevel: null,
  children: null,
  onToggle: _.noop()
};

SchemaCollapsePure.propTypes = {
  title: PropTypes.string,
  type: PropTypes.string,
  marginLeft: PropTypes.number,
  nestLevel: PropTypes.number,
  children: PropTypes.array,
  onToggle: PropTypes.func
};

const enhance = withStateHandlers(
  ({ initialCollapse = true }) => ({
    collapse: initialCollapse
  }),
  {
    onToggle: ({ collapse }) => () => ({ collapse: !collapse })
  }
);

export const SchemaCollapseWithState = enhance(SchemaCollapsePure);
export const SchemaCollapse = SchemaCollapseWithState;
