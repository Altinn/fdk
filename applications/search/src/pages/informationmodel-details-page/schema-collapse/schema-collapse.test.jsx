import React from 'react';
import { shallow } from 'enzyme';
import { SchemaCollapsePure } from './schema-collapse.component';

test('should render SchemaCollapsePure correctly with no props', () => {
  const wrapperWithNoProps = shallow(<SchemaCollapsePure />);
  expect(wrapperWithNoProps).toMatchSnapshot();
});

test('should render SchemaCollapsePure with props', () => {
  const defaultProps = {
    title: 'Tittel',
    type: 'Type',
    marginLeft: '16',
    nestLevel: 1
  };
  const wrapper = shallow(<SchemaCollapsePure {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
