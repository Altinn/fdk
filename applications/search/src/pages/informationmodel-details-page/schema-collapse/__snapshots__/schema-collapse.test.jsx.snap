// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SchemaCollapsePure correctly with no props 1`] = `""`;

exports[`should render SchemaCollapsePure with props 1`] = `
<React.Fragment>
  <React.Fragment>
    <button
      className="bg-transparent border-0 p-0 w-100"
      name="Tittel"
      style={
        Object {
          "marginLeft": "16",
        }
      }
    >
      <div
        className="d-flex align-items-center"
      >
        <div
          className="d-flex justify-content-start w-50"
        >
          <i
            className="fa font-weight-bold mr-2 fdk-color-blue-dark fa-angle-down"
          />
          <span
            className="fdk-text-extra-strong"
          >
            Tittel
          </span>
        </div>
        <span
          className="d-flex w-50"
        >
          Type
        </span>
      </div>
    </button>
    <Collapse
      appear={false}
      className="mt-2"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    />
  </React.Fragment>
</React.Fragment>
`;
